streams/src/test/java/org/apache/kafka/streams/integration/RegexSourceIntegrationTest.java

```diff
@@ -388,17 +387,24 @@ public void testNoMessagesSentExceptionFromOverlappingPatterns() throws Exceptio
         final StreamsBuilder builder = new StreamsBuilder();
 
 
-        // overlapping patterns here, no messages should be sent as TopologyBuilderException
+        // overlapping patterns here, no messages should be sent as TopologyException
         // will be thrown when the processor topology is built.
-
         final KStream<String, String> pattern1Stream = builder.stream(Pattern.compile("foo.*"));
         final KStream<String, String> pattern2Stream = builder.stream(Pattern.compile("f.*"));
```

Comment:
Should this line not throw with latest changes to check pattern overlap immediately?

Reply:
Note that before and now we only use `topicPattern.pattern().contains(otherPattern.pattern())` as we do not have an easy way to tell directly if two regex may overlap or not. So we still need to rely on runtime `SourceNodeFactory.getTopics` to detect eventually if there is some overlaps.

The check here that we moved from runtime is more like a pre-check which is not guaranteed 100 percent.